File: ios/Classes/Model/PPDfuPackageModel.swift
//
//  PPDfuPackageModel.swift
//  Pods
//
//  Created by lefu on 2025/4/3
//  


import Foundation


struct PPDfuPackageModel: Codable {
    let deviceSource: String
    let packageVersion: String
    let packages: PPDfuPackages
}

struct PPDfuPackages: Codable {
    let mcu: PPDfuPackageInfo?
    let ble: PPDfuPackageInfo?
    let res: PPDfuPackageInfo?
}

struct PPDfuPackageInfo: Codable {
    let filename: String
    let version: String
}

File: android/src/main/kotlin/com/example/pp_bluetooth_kit_flutter/model/PPDfuPackageModel.kt
package com.example.pp_bluetooth_kit_flutter.model

import com.google.gson.annotations.SerializedName




data class PPDfuPackageModel(
    @SerializedName("packages") val packages: PPDfuPackages,
    @SerializedName("deviceSource") val deviceSource: String,
    @SerializedName("packageVersion") val packageVersion: String,
)

data class PPDfuPackages(
    @SerializedName("mcu") val mcu: PPDfuPackageInfo?,
    @SerializedName("ble") val ble: PPDfuPackageInfo?,
    @SerializedName("res") val res: PPDfuPackageInfo?
)

data class PPDfuPackageInfo(
    @SerializedName("version") val version: String,
    @SerializedName("filename") val filename: String
)